google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/DefaultGoogleApiClientFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.resources;
+
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.appengine.v1.Appengine;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.cloud.tools.intellij.CloudToolsPluginInfoService;
+
+import com.intellij.openapi.components.ServiceManager;
+
+import org.jetbrains.annotations.Nullable;
+
+public class DefaultGoogleApiClientFactory extends GoogleApiClientFactory {
```

Comment:
The main value of this class is that it holds static (global) references to `NetHttpTransport` and `JacksonFactory` instances, which are required for instantiation of several different Google API clients. Both of these classes are thread-safe, and both actually recommend sharing global instances (see javadoc for [NetHttpTransport](https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/javanet/NetHttpTransport) and [JacksonFactory](https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/json/jackson/JacksonFactory) for efficiency.